src/python/pants/option/arg_splitter.py

```diff
@@ -109,73 +93,15 @@ def _get_known_goal_scopes(
                 yield alias, si
 
     def split_args(self, args: Sequence[str]) -> SplitArgs:
-        """Split the specified arg list.
+        native_split_args = self._native_arg_splitter.split_args(list(args))
 
-        args[0] is ignored.
-
-        Returns a SplitArgs tuple.
-        """
-        goals: OrderedSet[str] = OrderedSet()
-        specs: list[str] = []
-        passthru: list[str] = []
-        unknown_scopes: list[str] = []
         builtin_or_auxiliary_goal: str | None = None
+        canonical_goals = []
+        for goal in native_split_args.goals():
```

Comment:
Rust returns a string-valued goal name, here we get a ScopeInfo instance for it.